pkg/config/config.go

```diff
@@ -45,15 +45,18 @@ type OIDC struct {
 	FriendlyNameClaimName  string            `mapstructure:"friendly_name_claim_name"`
 }
 
-// LDAP holds configuration for authenticating on an LDAP server.
-type LDAP struct {
-	ServerEndpoint    string `mapstructure:"server_endpoint"`
-	BindDN            string `mapstructure:"bind_dn"`
-	BindPassword      string `mapstructure:"bind_password"`
-	DefaultUserGroup  string `mapstructure:"default_user_group"`
-	UsernameAttribute string `mapstructure:"username_attribute"`
-	UserBaseDN        string `mapstructure:"user_base_dn"`
-	UserFilter        string `mapstructure:"user_filter"`
+// RemoteAuthenticator holds remote authentication configuration.
+type RemoteAuthenticator struct {
+	// Enabled if set true will enable remote authentication
+	Enabled bool `mapstructure:"enabled"`
+	// BaseURL is the base URL of the remote authentication service (e.g. https://my-auth.example.com)
+	BaseURL string `mapstructure:"base_url" validate:"required"`
+	// AuthEndpoint is the endpoint to authenticate users (e.g. /auth)
+	AuthEndpoint string `mapstructure:"auth_endpoint" validate:"required"`
+	// DefaultUserGroup is the default group for the users authenticated by the remote service
+	DefaultUserGroup string `mapstructure:"default_user_group" validate:"required"`
+	// RequestTimeout time out for remote authentication requests
+	RequestTimeout time.Duration `mapstructure:"request_timeout" validate:"required"`
```

Comment:
I think that this should have a default and not be required.

Reply:
👍